### PR TITLE
Use chaff for handling fake requests

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-108

### CODE REVIEW

File: cmd/apiserver/main.go [hunk 1]
```diff
@@ -129,3 +135,7 @@ func apiKeyFunc() httplimit.KeyFunc {
 		return ipKeyFunc(r)
 	}
 }
+
+func handleChaff(tracker *chaff.Tracker, next http.Handler) http.Handler {
```
Comment (sethvargo) [inline cmd/apiserver/main.go:139]: I'm not a huge fan of one-line helpers, but the handler becomes really difficult to follow without it.

File: pkg/api/api.go [hunk 2]
```diff
@@ -96,16 +96,3 @@ type VerificationCertificateResponse struct {
 	Certificate string `json:"certificate"`
 	Error       string `json:"error"`
 }
-
-// CoverRequest is a client->server request for simulating traffic to the server.
```
Comment (sethvargo) [inline pkg/api/api.go:100]: Do we need to keep this in and mark as deprecated instead?

File: README.md [hunk 3]
```diff
@@ -130,58 +130,57 @@ keys and API keys have one of two levels of access: `DEVICE` or `ADMIN`.
...
-  ```
+The following APIs exist for the API server (`cmd/apiserver`). All APIs are JSON
+over HTTPS, only use `POST`, and require that the API key be passed in the HTTP
+header `X-API-Key`.
+
+In addition to "real" requests, the server also accept chaff (fake) requests.
+These can be used to obfuscate real traffic from a network observer or server
+operator. To initiate a chaff request, set the `X-Chaff` header on your request.
+The server will respond with a fake response that your client **MUST NOT**
+process. Client's should sporadically issue chaff requests.
```
Comment (mikehelmick) [inline README.md:141]: also - note that the client should send a request body that is of the appropriate size, but that that request body will not be processed at all.

File: README.md [hunk 4]
```diff
@@ -130,58 +130,57 @@ keys and API keys have one of two levels of access: `DEVICE` or `ADMIN`.
...
+
+    **VerifyCodeRequest:**
+
+    ```json
+    {
+      "code": "<the code>"
+    }
+    ```
+
+    **VerifyCodeResponse:**
```
Comment (mikehelmick) [inline README.md:154]: let's add a padding field here for the chaff requests, so the client knows to ignore it.

File: pkg/api/api.go (hunk 2 above)
Comment (mikehelmick) [reply pkg/api/api.go:100]: as far as I know , this isn't in use any anyone yet.

File: README.md (hunk 4 above)
Comment (sethvargo) [reply README.md:154]: Wouldn't the client have known it sent a chaff request? I think I'm missing why we'd need something in the response.

File: README.md (hunk 3 above)
Comment (sethvargo) [reply README.md:141]: Done

File: README.md (hunk 4 above)
Comment (mikehelmick) [reply README.md:155]: I mean, one would hope. We do need to pad out the responses so they're the right size, but it's also find to document that they're not JSON responses. Either way.

**Pull request merged**